src/components/tracks/trackItem.tsx

```diff
@@ -0,0 +1,80 @@
+import { Button } from "@mui/material";
+import { Track } from "@prisma/client";
+import DeleteIcon from "@mui/icons-material/Delete";
+import CloudUploadIcon from "@mui/icons-material/CloudUpload";
+import Image from "next/image";
+import { Utils } from "~/services/utils";
+
+type Props = {
+  track: Track;
+  currentTrackId: string | undefined;
+  index?: number;
+  updateFile?: (file: File, ref: string, id: string) => void;
+  deleteFile?: (id: string, ref: string) => void;
+  handleTrack?: (track: Track) => void;
+};
+
+export const TrackItem: React.FC<Props> = ({
+  track,
+  currentTrackId,
+  index,
+  deleteFile,
+  updateFile,
+  handleTrack,
+}) => {
+  const { id, ref, url, name, duration, singer } = track;
+  const showPlayingTrack = () => {
+    if (id === currentTrackId) {
+      return `bg-darkPrimary text-textRed`;
+    } else {
+      return `hover:bg-darkPrimary hover:text-textRed`;
+    }
+  };
+  return (
+    <div
+      key={id}
+      className={`flex flex-nowrap items-center gap-2 rounded-md  p-2 text-textSecondary transition duration-200 ease-in-out hover:cursor-pointer ${showPlayingTrack()} max-w-full`}
```

Comment:
It is better to use clsx + tailwind merge.

https://youtu.be/re2JFITR7TI?si=o0SkTTuWJRRRVgeB